src/code/mempak.c

```diff
@@ -1,184 +1,266 @@
+/**
+ * @file mempak.c
+ *
+ * This file implements an interface for a controller memory pak filesystem (pfs), with operations to create, delete,
+ * read, write and check the size of files, as well as obtain a listing of which files currently exist.
+ *
+ * Each file is assigned an uppercase ASCII letter as an identifier, the game name for each is marked as
+ * 'ZELDA DEMO TOOL', encoded according to the N64 Font Code described in section 26.3 of the N64 Programming Manual.
+ */
 #include "global.h"
 
+#define MEMPAK_MAX_FILES 10
+
 OSPfs sMempakPfsHandle;
 s32 sMempakFreeBytes;
-s32 sMempakFiles[10];
+s32 sMempakFiles[MEMPAK_MAX_FILES];
 
 u16 sMempakCompanyCode = 1;
 u32 sMempakGameCode = 1;
 
-// "ZELDA DEMO TOOL "
-u8 sMempakGameName[0x10] = { 0x33, 0x1E, 0x25, 0x1D, 0x1A, 0x0F, 0x1D, 0x1E,
-                             0x26, 0x28, 0x0F, 0x2D, 0x28, 0x28, 0x25, 0x0F };
-u8 sMempakExtName[8] = { 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00 };
-
-s32 Mempak_Init(s32 controllerNb) {
-    OSMesgQueue* mq;
+// Conversion from A-Z to N64 Font Code
+#define NCH(c) ((c)-0x27)
+// Conversion from spaces to N64 Font Code
+#define NCH_SPC (0x0F)
+
+u8 sMempakGameName[PFS_FILE_NAME_LEN] = { NCH('Z'), NCH('E'), NCH('L'), NCH('D'), NCH('A'), NCH_SPC,
+                                          NCH('D'), NCH('E'), NCH('M'), NCH('O'), NCH_SPC,  NCH('T'),
+                                          NCH('O'), NCH('O'), NCH('L'), NCH_SPC };
+
+u8 sMempakExtName[PFS_FILE_EXT_LEN] = { 0 };
+
+/**
+ * Initializes the memory pak filesystem for a memory pak in the controller plugged into the specified port. Subsequent
+ * memory pak operations will use the same controller port.
+ *
+ * @return true if the operation completed successfully, false otherwise
+ */
+s32 Mempak_Init(s32 controllerNo) {
+    OSMesgQueue* serialEventQueue;
     s32 pad;
     s32 ret = false;
 
-    mq = PadMgr_AcquireSerialEventQueue(&gPadMgr);
+    serialEventQueue = PadMgr_AcquireSerialEventQueue(&gPadMgr);
 
-    if (!osPfsInitPak(mq, &sMempakPfsHandle, controllerNb)) {
+    if (osPfsInitPak(serialEventQueue, &sMempakPfsHandle, controllerNo) == 0) {
         ret = true;
     }
 
     osPfsFreeBlocks(&sMempakPfsHandle, &sMempakFreeBytes);
-    PadMgr_ReleaseSerialEventQueue(&gPadMgr, mq);
-
+    PadMgr_ReleaseSerialEventQueue(&gPadMgr, serialEventQueue);
     return ret;
 }
 
-s32 Mempak_GetFreeBytes(s32 controllerNb) {
+s32 Mempak_GetFreeBytes(s32 controllerNo) {
     return sMempakFreeBytes;
 }
 
-s32 Mempak_FindFile(s32 controllerNb, char start, char end) {
-    OSMesgQueue* mq;
+/**
+ * Checks if the files identified by letters between `start` and `end` (inclusive) exist on the memory pak.
+ *
+ * @param controllerNo Unused, the controller used is that which was last passed to `Mempak_Init`
```

Comment:
I suppose this is libultra terminology, rather than using `num`?

Reply:
done